components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -456,100 +459,108 @@ func InjectCertConfig(notebook *nbv1.Notebook, configMapName string) error {
-                                            jTime := items[j].(map[string]interface{})["created"].(string)
-                                            return iTime > jTime // Lexicographical comparison of RFC3339 timestamps
-                                        })
-                                        imageHash := items[0].(map[string]interface{})["dockerImageReference"].(string)
-                                        notebook.Spec.Template.Spec.Containers[0].Image = imageHash
-                                        // Update the JUPYTER_IMAGE environment variable
-                                        for i, envVar := range notebook.Spec.Template.Spec.Containers[0].Env {
-                                            if envVar.Name == "JUPYTER_IMAGE" {
-                                                notebook.Spec.Template.Spec.Containers[0].Env[i].Value = imageHash
-                                                break
-                                            }
-                                        }
-                                        imagestreamFound = true
-                                        break
-                                    }
-                                }
-                            }
-                        }
-                    }
-                    if imagestreamFound {
-                        break
-                    }
-                }
-
-                if !imagestreamFound {
-                    log.Info("Imagestream not found in any of the specified namespaces", "imagestreamName", imagestreamName, "tag", tag)
-                }
-            }
-        }
-    }
-
-    return nil
+	// Create a dynamic client
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		log.Error(err, "Error creating dynamic client")
+		return err
+	}
+	// Specify the GroupVersionResource for imagestreams
+	ims := schema.GroupVersionResource{
+		Group:    "image.openshift.io",
+		Version:  "v1",
+		Resource: "imagestreams",
+	}
+
+	annotations := notebook.GetAnnotations()
+	if annotations != nil {
+		if imageSelection, exists := annotations["notebooks.opendatahub.io/last-image-selection"]; exists {
+
+			containerFound := false
+			// Iterate over containers to find the one matching the notebook name
+			for i, container := range notebook.Spec.Template.Spec.Containers {
+				if container.Name == notebook.Name {
+					containerFound = true
+
+					// Check if the image selection has an internal registry, if so  will pickup this. This value constructed on the initialization of the Notebook CR.
+					if strings.Contains(container.Image, "image-registry.openshift-image-registry.svc:5000") {
+						log.Info("Internal registry found. Will pickup the default value from image field.")
+						return nil
+					} else {
+						log.Info("No Internal registry found, pick up imageHash from status.tag.dockerImageReference")
```

Comment:
This is info message, let's be precise here:

```suggestion
						log.Info("No internal registry found, let's pick up image reference from relevant ImageStream 'status.tags[].tag.dockerImageReference'")
```

Reply:
done